gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -104,14 +104,13 @@ public final class ConjurePlugin implements Plugin<Project> {
     @Override
     public void apply(Project project) {
         project.getPlugins().apply(BasePlugin.class);
-        project.getPluginManager().apply(ConjureBasePlugin.class);
+        ConjureBasePlugin conjureBasePlugin = project.getPlugins().apply(ConjureBasePlugin.class);
 
-        TaskProvider<?> compileIrTask = project.getTasks().named(ConjureBasePlugin.COMPILE_IR_TASK);
+        TaskProvider<?> compileIrTask = conjureBasePlugin.compileIrProvider();
         TaskProvider<GenerateConjureServiceDependenciesTask> serviceDependencyTask = project.getTasks()
                 .named(ConjureBasePlugin.SERVICE_DEPENDENCIES_TASK, GenerateConjureServiceDependenciesTask.class);
 
-        ConjureExtension conjureExtension =
-                project.getExtensions().create(ConjureExtension.EXTENSION_NAME, ConjureExtension.class);
+        ConjureExtension conjureExtension = conjureBasePlugin.conjureExtension();
```

Comment:
I moved the `ConjureExtension` registration to `ConjureBasePlugin`, and updated the interacting pieces to pass references from the plugin registration rather than digging through get/find-by-X. It seemed cleaner, but I wouldn't be surprised if it's an unspeakable sin against gradle.

Reply:
Hmm, looks good to me but actually not sure why this isn't applied more often. I guess it requires a tighter coupling between plugins but that seems fine in this case. Maybe @CRogers has a stronger opinion here :)

Reply:
I have done both. I don't know how I feel about one versus the other. The just access the plugin method feels nicer from a general-java-coding standpoint, everything in Gradle land does the latter though.